python/benchmark/gen_data_distributed.py

```diff
@@ -0,0 +1,554 @@
+#
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import random
+import sys
+from abc import abstractmethod
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from pyspark.mllib.random import RandomRDDs
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.functions import array
+from sklearn.datasets import (
+    make_blobs,
+    make_classification,
+    make_low_rank_matrix,
+    make_regression,
+)
+
+from benchmark.utils import WithSparkSession, inspect_default_params_from_func, to_bool
+
+
+def dtype_to_pyspark_type(dtype: Union[np.dtype, str]) -> str:
+    """Convert np.dtype to the corresponding pyspark type"""
+    dtype = np.dtype(dtype)
+    if dtype == np.float32:
+        return "float"
+    elif dtype == np.float64:
+        return "double"
+    else:
+        raise RuntimeError("Unsupported dtype, found ", dtype)
+
+
+class DataGen(object):
```

Comment:
Same here. Try to import DataGen class from gen_data.py to avoid duplicate implementations.